src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -40,17 +40,10 @@ internal static Encoding GetEncoding(string text)
         /// </summary>
         internal static void ReadBytes(Stream stream, byte[] buffer, int bytesToRead)
         {
-            int bytesLeftToRead = bytesToRead;
-
-            int totalBytesRead = 0;
-
-            while (bytesLeftToRead > 0)
+            int bytesRead = stream.ReadAtLeast(buffer.AsSpan(0, bytesToRead), bytesToRead, throwOnEndOfStream: false);
+            if (bytesRead < bytesToRead)
             {
-                int bytesRead = stream.Read(buffer, totalBytesRead, bytesLeftToRead);
-                if (bytesRead == 0) throw new IOException(SR.UnexpectedEndOfStream);
-
-                totalBytesRead += bytesRead;
-                bytesLeftToRead -= bytesRead;
+                throw new IOException(SR.UnexpectedEndOfStream);
```

Comment:
We don't want to use ReadExactly here because we want a different exception message?  Otherwise it would throw an EndOfStreamException which derives from IOException, right?

Reply:
Correct. Both the exception message and Type would be changed if we used ReadExactly here. I didn't want to change that behavior.